pkg/kubectl/cmd/resource/get.go

```diff
@@ -368,31 +378,21 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
 			}
 		}
 
-		if shouldGetNewPrinterForMapping(printer, lastMapping, mapping) {
-			if printer != nil {
-				w.Flush()
-			}
+		if shouldGetNewPrinterForMapping(lastMapping, mapping) {
+			w.Flush()
 
-			printWithNamespace := options.AllNamespaces
-			if mapping != nil && mapping.Scope.Name() == meta.RESTScopeNameRoot {
-				printWithNamespace = false
-			}
+			// TODO: should AllNamespaces be part of PrintFlags?
+			//printWithNamespace := options.AllNamespaces
+			//if mapping != nil && mapping.Scope.Name() == meta.RESTScopeNameRoot {
+			//	printWithNamespace = false
+			//}
 
-			printOpts := cmdutil.ExtractCmdPrintOptions(cmd, printWithNamespace)
+			// TODO: this should be moved to the PrintGeneric method
 			// if cmd does not specify output format and useOpenAPIPrintColumnFlagLabel flag is true,
 			// then get the default output options for this mapping from OpenAPI schema.
-			if !cmdSpecifiesOutputFmt(cmd) && useOpenAPIPrintColumns {
-				updatePrintOptionsForOpenAPI(f, mapping, printOpts)
-			}
-
-			printer, err = cmdutil.PrinterForOptions(printOpts)
-			if err != nil {
-				if !errs.Has(err.Error()) {
-					errs.Insert(err.Error())
-					allErrs = append(allErrs, err)
-				}
-				continue
-			}
+			//if !cmdSpecifiesOutputFmt(cmd) && useOpenAPIPrintColumns {
+			//	updatePrintOptionsForOpenAPI(f, mapping, printOpts)
```

Comment:
@soltysh Ah, forgot that this will be moving to the server anyway. So we might not even need to worry about this.

I am still a bit hesitant about having this command go straight to `cmdutil.PrintObject` for printing. It might make more sense to have this be one of the few (only?) command that instantiates PrintFlags directly and deal with a printer.